cdisc_rules_engine/services/define_xml/base_define_xml_reader.py

```diff
@@ -286,33 +267,11 @@ def _get_variable_datatype(self, data_type):
         }
         return variable_type_map.get(data_type, data_type)
 
-    # TODO: subclass define.xml versions
-    # https://github.com/cdisc-org/cdisc-rules-engine/issues/390
     def _get_origin_type(self, itemdef):
-        if not itemdef.Origin:
-            return
-        version_origin_map = {
-            DefineXMLVersions.DEFINE_XML_2_0_0: lambda origin: origin.Type,
-            DefineXMLVersions.DEFINE_XML_2_1_0: lambda origin: origin[0].Type,
-        }
-        return version_origin_map.get(DefineXMLVersions(self.get_define_version()))(
-            itemdef.Origin
-        )
+        return itemdef.Origin[0].Type if itemdef.Origin else None
```

Comment:
NIT: I think we should keep version specific implementations out of the base class